packages/core/src/prompts/group-multiselect.ts

```diff
@@ -0,0 +1,70 @@
+import Prompt, { PromptOptions } from './prompt';
+
+interface GroupMultiSelectOptions<T extends { value: any }> extends PromptOptions<GroupMultiSelectPrompt<T>> {
+	options: Record<string, T[]>;
+	initialValues?: T['value'][];
+	required?: boolean;
+	cursorAt?: T['value'];
+}
+export default class GroupMultiSelectPrompt<T extends { value: any }> extends Prompt {
+	options: (T & { group: string | boolean })[];
+	cursor: number = 0;
+
+	getGroupItems(group: string): T[] {
+		return this.options.filter(i => i.group === group);
+	}
+
+	isGroupSelected(group: string) {
+		const items = this.getGroupItems(group);
+		return items.every(i => this.value.includes(i.value));
+	}
+
+	private toggleValue() {
+		const item = this.options[this.cursor];
+		if (item.group === true) {
+			const group = item.value;
+			const groupedItems = this.getGroupItems(group);
+			if (this.isGroupSelected(group)) {
+				this.value = this.value.filter((v: string) => groupedItems.findIndex(i => i.value === v) === -1);
+			} else {
+				this.value = [...this.value, ...groupedItems.map(v => v.value)];
+			}
+			this.value = Array.from(new Set(this.value));
+		} else {
+			const selected = this.value.includes(item.value);
+			this.value = selected
+				? this.value.filter((value: T['value']) => value !== item.value)
+				: [...this.value, item.value];
+		}
+	}
+
+	constructor(opts: GroupMultiSelectOptions<T>) {
+		super(opts, false);
+
+		this.options = Object.keys(opts.options).reduce((acc, key) => {
+			acc.push({ value: key, group: true, label: key }, ...opts.options[key].map(opt => ({ ...opt, group: key })));
+			return acc;
+		}, [] as { value: any, [key: string]: any }[]);
+		this.value = [...(opts.initialValues ?? [])];
+		this.cursor = Math.max(
+			this.options.findIndex(({ value }) => value === opts.cursorAt),
+			0
+		);
+
+		this.on('cursor', (key) => {
+			switch (key) {
+				case 'left':
+				case 'up':
+					this.cursor = this.cursor === 0 ? this.options.length - 1 : this.cursor - 1;
```

Comment:
```suggestion
					this.cursor = (this.options.length + (cursor - 1)) % this.options.length;
```

Reply:
This feels a little too clever for me, to be honest.